packages/python-packages/api-stub-generator/apistub/nodes/_argtype.py

```diff
@@ -10,15 +10,25 @@
 
 class ArgType:
     """Represents Argument type
-    :param str name: Name of the argument
-    :param str argtype: Type of the argument. for e.g. str, int, BlobBlock
-    :param str default: Default value for the argument, If any
+    :param str name: Name of the argument.
+    :param str argtype: Type of the argument (e.g. str, int, BlobBlock).
+    :param str default: Default value for the argument, If any.
+    :param str keyword: The keyword for the arg type.
+    :param FunctionNode func_node: The function node this belongs to.
     """
-
-    def __init__(self, name, argtype=None, default=None, func_node = None):
+    def __init__(self, name, *, argtype, default, keyword, func_node=None):
         self.argname = name
-        self.argtype = argtype
-        self.default = default
+        if default == inspect._empty or default == inspect.Parameter.empty:
+            self.is_required = True
+            self.default = None
+        else:
+            self.is_required = False
+            self.default = str(default) if default is not None else "..."
```

Comment:
if a default exists and it is an object, should we call str() on the `__name__` of the class?

Reply:
This is surprisingly tricky. It wouldn't be the name of the class--it would need to be the actual string representation of the default declaration. Otherwise it would be confused with a default that is a type rather than an object.

Right now it will produce something like `< class Foo instance at HEX >` which isn't ideal, but is somewhat understandable. Do you know of an instance where an object is being used as a default?

Reply:
Ah I see. I can only think of where we might set the default to an enum like here: https://github.com/Azure/azure-sdk-for-python/blob/main/sdk/appconfiguration/azure-appconfiguration/azure/appconfiguration/_azure_appconfiguration_client.py#L214

Which I imagine will work fine with str(). So maybe this is not an edge case we need to worry about right now.

Reply:
Right, enum was the most likely case I thought of too. I added it to the test and it does work :)